### PR TITLE
Update docs in README.md

### PR DESCRIPTION
**Pull request by cham11ng**
Update docs in README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,99 @@
 [![LICENSE](https://img.shields.io/github/license/leapfrogtechnology/async-store.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store/blob/master/LICENSE)
 [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store)
 
-Global store utility for an async operation lifecycle and chain of callbacks. 
+Global store utility for an async operation lifecycle and chain of callbacks. It is utility tool similar to [continuation-local-storage](https://github.com/othiym23/node-continuation-local-storage) which allows us to set and get values that are scoped to the lifetime of these chains of callbacks. It uses [domain](https://nodejs.org/api/domain.html) native Node.js module.
+
+**Simple example,**
```
Comment (kabirbaidhya) [inline README.md]: Remove comma and the bold formatting; convert it to a sub-heading instead.

File: README.md [hunk 2]
```diff
@@ -5,7 +5,99 @@
 [![LICENSE](https://img.shields.io/github/license/leapfrogtechnology/async-store.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store/blob/master/LICENSE)
 [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store)
 
-Global store utility for an async operation lifecycle and chain of callbacks. 
+Global store utility for an async operation lifecycle and chain of callbacks. It is utility tool similar to [continuation-local-storage](https://github.com/othiym23/node-continuation-local-storage) which allows us to set and get values that are scoped to the lifetime of these chains of callbacks. It uses [domain](https://nodejs.org/api/domain.html) native Node.js module.
```
Comment (kabirbaidhya) [inline README.md]: Move the domain thing to the next line in **bold** font as a **Note**.

File: README.md [hunk 3]
```diff
@@ -5,7 +5,99 @@
...
+
+**Simple example,**
+
+```js
+var asyncStore = require('async-store');
```
Comment (kabirbaidhya) [inline README.md]: const

Comment (kabirbaidhya) [reply README.md]: and the correct package name it's `@leapfrogtechnology/async-store` check the npm registry link.

File: README.md [hunk 4]
```diff
@@ -5,7 +5,99 @@
...
+
+var callback = function {
```
Comment (kabirbaidhya) [inline README.md]: Could be either an arrow function `=>` if named in this way as a variable or should be a `function` only if it's defined as a function.

File: README.md [hunk 5]
```diff
@@ -5,7 +5,99 @@
...
+    .then(() => {
+      // Store value is also available at the end of the promise chain.
+      console.log('Value of foo: ', globalStore.get('foo'));
+    });
+};
+
+asyncStore.initialize()(callback);
+```
+
+**Simple express.js example or [here](https://github.com/kabirbaidhya/async-store-sample)**
```
Comment (kabirbaidhya) [inline README.md]: 1. Heading - `Express Example`
2. Move the other example in a different section `More Examples` under which you can just put a link to the source.

File: README.md [hunk 6]
```diff
@@ -5,7 +5,99 @@
...
+app.listen(port, () => console.log(`Example app listening on port ${port}!`));
+```
+
+### initializeMiddleware()
+
+Middleware to initialize the async store and make it accessible from all the subsequent middlewares or async operations triggered afterwards.
+
+```js
+var express = require('express');
+var asyncStore = require('async-store');
```
Comment (kabirbaidhya) [inline README.md]: Let's change all the `asyncStore` to `store` in the examples.

File: README.md [hunk 7]
```diff
@@ -5,7 +5,117 @@
...
+  res.json({ message: 'Hello World' });
+});
+
+app.listen(port, () => console.log(`Example app listening on port ${port}!`));
+```
+
+### Other Example
+
+- [Async Store Sample](https://github.com/kabirbaidhya/async-store-sample)
+
```
Comment (kabirbaidhya) [inline README.md]: Should have a heading - `## API Docs`

File: README.md [hunk 8]
```diff
@@ -5,7 +5,117 @@
 [![LICENSE](https://img.shields.io/github/license/leapfrogtechnology/async-store.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store/blob/master/LICENSE)
 [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store)
 
-Global store utility for an async operation lifecycle and chain of callbacks. 
+Global store utility for an async operation lifecycle and chain of callbacks. It is utility tool similar to [continuation-local-storage](https://github.com/othiym23/node-continuation-local-storage) which allows us to set and get values that are scoped to the lifetime of these chains of callbacks.
```
Comment (mesaugat) [inline README.md]:
```diff
+ It is a utility
- It is utility
```

File: README.md [hunk 9]
```diff
@@ -5,7 +5,117 @@
...
+
+**It uses [domain](https://nodejs.org/api/domain.html) native Node.js module.**
```
Comment (mesaugat) [inline README.md]: **async-store uses Node.js [domain](https://nodejs.org/api/domain.html) module under the hood.**

File: README.md [hunk 10]
```diff
@@ -5,7 +5,157 @@
...
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      // Store value is also available at the end of the promise chain.
```
Comment (mesaugat) [inline README.md]: Remove `also`

File: README.md [hunk 11]
```diff
@@ -5,7 +5,157 @@
...
+
+  Promise.resolve()
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      // Store value is also available at the end of the promise chain.
```
Comment (mesaugat) [inline README.md]: Same.

File: README.md [hunk 12]
```diff
@@ -5,7 +5,157 @@
...
+
+```js
+store.set({ foo: 'Hello', bar: 'World' });
+```
+
+### get()
+
+It gets a value by a key from the store.
+
+- `{string} key` - Key specifies property of store.
```
Comment (mesaugat) [inline README.md]: Missing`@params`

File: README.md [hunk 13]
```diff
@@ -5,7 +5,157 @@
...
+
+```js
+store.get('foo');
+```
+
+### find()
+
+It gets a value by a key from the store. If anything fails, it returns null without emitting error event.
+
+- `{string} key` - Key specifies property of store.
```
Comment (mesaugat) [inline README.md]: .

**Pull request merged**